enterprise/api/v1/views/enterprise_group.py

```diff
@@ -37,13 +46,158 @@ def get_queryset(self, **kwargs):
+                                    defaults={'active': True},
+                                )
+                                enterprise_learners.append(
+                                    models.EnterpriseGroupMembership(
+                                        enterprise_customer_user=enterprise_customer_user,
+                                        group=group,
+                                    )
+                                )
+                                total_existing_users_processed += 1
+                            except User.DoesNotExist:
+                                pending_ecu, __ = models.PendingEnterpriseCustomerUser.objects.update_or_create(
+                                    user_email=user_email,
+                                    enterprise_customer=group.enterprise_customer,
+                                )
+                                new_pending_learners.append(
+                                    models.EnterpriseGroupMembership(
+                                        pending_enterprise_customer_user=pending_ecu,
+                                        group=group,
+                                    )
+                                )
+                                total_new_users_processed += 1
+
+                    objs = models.EnterpriseGroupMembership.objects.bulk_create(
+                        objs=enterprise_learners + new_pending_learners,
+                        ignore_conflicts=True,
+                    )
+                    total_records_processed += len(objs)
+            data = {
+                'records_processed': total_records_processed,
+                'new_learners': total_new_users_processed,
+                'existing_learners': total_existing_users_processed,
+            }
+            return Response(data, status=201)
+        return Response(data="Error: missing request data: `learner_emails`.", status=400)
+
+    @action(methods=['post'], detail=False, permission_classes=[permissions.IsAuthenticated])
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, group_uuid: get_enterprise_customer_from_enterprise_group_id(group_uuid)
+    )
+    def remove_learners(self, request, group_uuid):
+        """
+        POST /enterprise/api/v1/enterprise-group/<group uuid>/remove_learners
+
+        Required Arguments:
+            - ``learner_emails``:
+                List of learner emails to associate with the group.
+
+        Returns:
+            - ``records_deleted``:
+                Number of membership records removed
+        """
+        try:
+            group = self.get_queryset().get(uuid=group_uuid)
+        except models.EnterpriseGroup.DoesNotExist as exc:
+            raise Http404 from exc
+        if requested_emails := request.POST.dict().get('learner_emails'):
+            records_deleted = 0
+            for user_email_batch in utils.batch(requested_emails.split(','), batch_size=200):
```

Comment:
The ticket specifies a request limit of 200 users per call, are we going to enforce that?

Reply:
Updated requirements, the ticket is out of date, I'll update!